src/string_id_null_ids.cpp

```diff
@@ -46,6 +46,7 @@ MAKE_NULL_ID2( trap, "tr_null" )
 MAKE_NULL_ID2( construction_category, "NULL", -1 )
 MAKE_NULL_ID2( ammo_effect, "AE_NULL", 0 )
 MAKE_NULL_ID2( field_type, "fd_null", 0 )
+MAKE_NULL_ID2( move_mode, "MM_NULL", 0 )
```

Comment:
i don't think we really use null ids anymore? i've made several generic_factory things and have never needed to use it.

Reply:
Yes, null ids are terrible. If you need a variable that can contain either an id or nothing, wrap the id into `std::optional`.

With null ids, you never know where a null id is allowed / supported / will trigger some special behaviour.